torch/functional.py

```diff
@@ -430,6 +431,78 @@ def stft(input, n_fft, hop_length=None, win_length=None, window=None,
     return _VF.stft(input, n_fft, hop_length, win_length, window, normalized, onesided)
 
 
+def istft(input, n_fft, hop_length=None, win_length=None, window=None,
+          center=True, normalized=False, onesided=True, length=None):
+    # type: (Tensor, int, Optional[int], Optional[int], Optional[Tensor], bool, bool, bool, Optional[int]) -> Tensor
+    r"""Inverse short time Fourier Transform. This is expected to be the inverse of :func:`~torch.stft`.
+    It has the same parameters (+ additional optional parameter of :attr:`length`) and it should return the
+    least squares estimation of the original signal. The algorithm will check using the NOLA condition (
+    nonzero overlap).
+
+    Important consideration in the parameters :attr:`window` and :attr:`center` so that the envelop
+    created by the summation of all the windows is never zero at certain point in time. Specifically,
+    :math:`\sum_{t=-\infty}^{\infty} w^2[n-t\times hop\_length] \cancel{=} 0`.
+
+    Since :func:`~torch.stft` discards elements at the end of the signal if they do not fit in a frame,
+    ``istft`` may return a shorter signal than the original signal (can occur if :attr:`center` is False
+    since the signal isn't padded).
+
+    If :attr:`center` is ``True``, then there will be padding e.g. ``'constant'``, ``'reflect'``, etc.
+    Left padding can be trimmed off exactly because they can be calculated but right padding cannot be
+    calculated without additional information.
+
+    Example: Suppose the last window is:
+    ``[17, 18, 0, 0, 0]`` vs ``[18, 0, 0, 0, 0]``
+
+    The :attr:`n_fft`, :attr:`hop_length`, :attr:`win_length` are all the same which prevents the calculation
+    of right padding. These additional values could be zeros or a reflection of the signal so providing
+    :attr:`length` could be useful. If :attr:`length` is ``None`` then padding will be aggressively removed
+    (some loss of signal).
+
+    [1] D. W. Griffin and J. S. Lim, "Signal estimation from modified short-time Fourier transform,"
+    IEEE Trans. ASSP, vol.32, no.2, pp.236-243, Apr. 1984.
+
+    Arguments:
+        input (Tensor): The input tensor. Expected to be output of :func:`~torch.stft` where
+            each row of a channel is a frequency and each column is a window.
+            It has a size of either (..., ``fft_size``, ``n_frame``, 2)
```

Comment:
Here, I would consider updating the documentation to not have a triple dot, and only talk about 4d tensors. thoughts?